iothub/service/src/Exceptions/ExceptionHandlingHelper.cs

```diff
@@ -18,181 +16,131 @@ internal class ExceptionHandlingHelper
+        // There are two things to consider when surfacing service errors to the user, the 6-digit error code and
+        // the error description. Ideally, when a backend service returns an error, both of these fields are set
+        // in the same place. However, IoT hub is returning the 6-digit code in the response content, while
+        // the error description in the response header. Therefore, there is a chance that the 6-digit error code
+        // does not match the error description. For that reason, the SDK will do its best to decide what to surface to the user.
+        // The SDK will attempt to retrieve the integer error code from the response content and the error description
+        // from the response header. Through a 'description' to 'error code' enum mapping, the SDK will check if
+        // both values are a match. If so, the SDK will populate the exception with the proper Code. In the case where
+        // there is a mismatch between the error code and the description, the SDK returns
+        // IotHubStatusCode.Unknown and log a warning.
+        internal static async Task<IotHubErrorCode> GetIotHubErrorCodeAsync(HttpResponseMessage response)
         {
-            // First we will attempt to retrieve the error code from the response content.
-            string responseContentStr = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
-
-            // There are two things to consider when surfacing service errors to the user, the 6-digit error code and the code description. Ideally, when a backend service
-            // returns an error, both of these fields are set in the same place. However, IoT hub is returning the 6-digit code in the response content, while
-            // the error description in the response header. Therefore, there is a chance that the 6-digit error code does not match the error description. For that reason,
-            // the SDK will do its best to decide what to surface to the user.
-            // The SDK will attempt to retrieve the integer error code from the response content and the error description from the response header. Through a 'description'
-            // to 'error code' enum mapping, the SDK will check if both values are a match. If so, the SDK will populate the exception with the proper Code. In the case where
-            // there is a mismatch between the error code and the description, the SDK returns ErrorCode.InvalidErrorCode and log a warning.
-
-            int errorCodeValue = (int)ErrorCode.InvalidErrorCode;
+            string responseBody = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
+            IoTHubExceptionResult responseContent = null;
             try
             {
-                IoTHubExceptionResult responseContent = JsonConvert.DeserializeObject<IoTHubExceptionResult>(responseContentStr);
+                responseContent = JsonConvert.DeserializeObject<IoTHubExceptionResult>(responseBody);
```

Comment:
We should be able to improve this type so that the various fields of this object are defined, and we can access the property we want directly, rather than having to deserialize again to a dictionary (below).